### ISSUE TITLE
Option to support clippy's semicolon_if_nothing_returned

### ISSUE THREAD

**Issue by nstinus** [rust-lang/rustfmt#5355]
Hi,

Is there an option to properly format code to adhere to https://rust-lang.github.io/rust-clippy/beta/index.html#semicolon_if_nothing_returned?

Thanks !

**Comment by calebcartwright** [comment]
No, not unless I'm forgetting something. However, it's important to keep in mind that rustfmt is not designed/intended to satiate clippy's rules nor vice versa, so it's not really a surprise.

rustfmt strictly operates at an earlier phase (AST pre-expansion, lowering, etc.) but we may have enough info available to support such an option.

If anyone is sufficiently interested then certainly feel free to submit a PR, but I don't see this being something that the rustfmt team will ever work on due to priority/bandwidth constraints.

**Comment by nstinus** [comment]
Understood. I think this could be useful as it would adhere to the recommended [style](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/statements.md#expressions-in-statement-position).
Could you share some pointers? I'd be happy to take a look if no-one more qualify can.

**Comment by calebcartwright** [comment]
I suspect the problem that you're going to run into is that rustfmt isn't going to have the information needed to be able to safely make that determination due the level it operates at as I mentioned earlier.

Not sure off hand where to best point you, but perhaps inspecting where the [`trailing_semicolon`](https://rust-lang.github.io/rustfmt/?version=v1.4.38&search=#trailing_semicolon) option is utilized, as well as how/where `is_last_expr()` helper function to see if there's anything there that helps
